Source/DiabloUI/dialogs.cpp

```diff
@@ -168,7 +168,7 @@ void Init(const char *text, const char *caption, bool error, bool renderBehind)
 		vecOkDialog.push_back(new UiText(text, rect2, UIS_CENTER));
 
 		SDL_Rect rect3 = { (Sint16)(PANEL_LEFT + 265), (Sint16)(UI_OFFSET_Y + 265), SML_BUTTON_WIDTH, SML_BUTTON_HEIGHT };
-		vecOkDialog.push_back(new UiButton(&SmlButton, _("OK"), &DialogActionOK, rect3, 0));
+		vecOkDialog.push_back(new UiButton(&SmlButton, _("OK").c_str(), &DialogActionOK, rect3, 0));

```

Comment:
The underscore returns an `std::string` now, which is destroyed after this call, so it points to deleted memory.

Reply:
In that case, yes, `UiButton` should probably store `m_text` as a `std::string`.

There are probably other cases like this, and they are non-trivial to detect, which makes this a bit complicated.

Reply:
As I mentioned in the other comment, `_` should only return the key, not the formatted string. In which case the return type can be `const std::string &` or `const char *`.